core/src/test/java/org/apache/iceberg/expressions/TestInclusiveMetricsEvaluatorWithExtract.java

```diff
@@ -683,4 +787,802 @@ public void testIntegerNotIn() {
         .as("Should read: id above upper bound (85 > 79, 86 > 79)")
         .isTrue();
   }
+
+  private static Stream<Arguments> timestampLtParameters() {
+    return Stream.of(
+            Arguments.of(
+                    "$.event_timestamp_nano_long",
+                    Types.TimestampNanoType.withZone().toString(),
+                    L_TS_NANO_MIN_VALUE_MINUS_10,
+                    TS_NANO_MIN_VALUE,
+                    TS_NANO_MAX_VALUE,
+                    "$.event_timestamp_nano_ts_type",
+                    TS_COMMON_MIN_VALUE_MINUS_10.toString(),
+                    TS_COMMON_MIN_VALUE.toString(),
+                    TS_COMMON_MAX_VALUE.toString()),
+            Arguments.of(
+                    "$.event_timestamp_long",
+                    Types.TimestampType.withZone().toString(),
+                    L_TS_MIN_VALUE_MINUS_10,
+                    TS_MIN_VALUE,
+                    TS_MAX_VALUE,
+                    "$.event_timestamp_ts_type",
+                    TS_COMMON_MIN_VALUE_MINUS_10.toString(),
+                    TS_COMMON_MIN_VALUE.toString(),
+                    TS_COMMON_MAX_VALUE.toString()),
+            Arguments.of(
+                    "$.event_time_long",
+                    Types.TimeType.get().toString(),
+                    L_TIME_MIN_VALUE_MINUS_10,
+                    TIME_MIN_VALUE,
+                    TIME_MAX_VALUE,
+                    "$.event_time_ts_type",
+                    T_TIME_MIN_VALUE_MINUS_10,
+                    T_TIME_MIN_VALUE,
+                    T_TIME_MAX_VALUE));
+  }
+
+  @ParameterizedTest
+  @MethodSource("timestampLtParameters")
+  public void testTimestampLt(
```

Comment:
Testing different expression types is already done by other tests, so the tests for this feature should use a simple predicate (like equals only) and go through different (Iceberg type, Variant type) pairs.